scripts/gen

```diff
@@ -25,43 +25,56 @@ if [[ "$1" == "validate" ]] && [[ "$2" == "--cluster-hardware" ]]; then
     exit
 fi
 
-if [[ "$1" == "deploy" ]]; then
-    if [[ "$2" == "--inv-add-ports-ipmi" ]] || \
-        [[ "$2" == "--inv-add-ports-pxe" ]] || \
-        [[ "$2" == "--create-inventory" ]] || \
-        [[ "$2" == "--install-cobbler" ]] || \
-        [[ "$2" == "--help" ]] || [[ "$2" == "-h" ]]; then
-        gen.py $@
-        exit
+if [[ "$1" == "deploy" ]] && ([[ "$2" == "-a" ]] || [[ "$2" == "--all" ]] || \
+            [[ "$2" == "-p" ]] || [[ "$2" == "-f" ]] || [ -z "$2" ]); then
+        shift
+
+    if [[ "$1" == "-a" ]] || [[ "$1" == "--all" ]]; then
+        shift
     fi
-fi
 
-if [[ "$1" == "deploy" ]]; then
-    gen.py validate --config-file
+    gen.py validate --config-file $@
     if [ $? -ne 0 ]; then
```

Comment:
We could replace all of these `if [ $? -ne 0]; then` conditionals with a single `trap 'exit' ERR` at the top. (Just mentioning this even though it's not really related to your patch).